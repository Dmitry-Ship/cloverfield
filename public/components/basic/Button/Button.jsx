import React, { PropTypes } from 'react';

import styles from './Button.styl'

import Icon from '../Icon';

const Button = ({ type,
                  label,
                  onClick,
                  className,
                  kind,
                  name,
                  disabled,
                  children }) => {
  return (
    <button
      type={type}
      onClick={onClick}
      disabled={disabled}
      className={`${styles[kind]} ${className}`} >
      <div>
        <span>{label}</span>
        {name && <Icon name={name} />}
      </div>
    </button>
  )
}

export default Button;

Button.defaultProps = {
  disabled: false,
  kind: 'primary'
}

Button.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.any,
  icon: PropTypes.string,

}
