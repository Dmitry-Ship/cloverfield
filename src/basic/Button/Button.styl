.button
  border none
  size 85px 45px
  font-size 1rem
  cursor pointer
  transition: 300ms ease-in-out
  &--primary
    background $color__secondary
    color white
    &:hover
      background $color__secondary--light
    &:active
      background $color__secondary--dark
  &--secondary
    background transparent
    border 3px solid $color__secondary
    color $color__secondary
    &:hover
      border 3px solid $color__secondary--light
      color $color__secondary--light
    &:active
      border 3px solid $color__secondary--dark
      color $color__secondary--dark
      background transparent
