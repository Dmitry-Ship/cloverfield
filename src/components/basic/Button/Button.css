@import '../../../stylesheets/colors.css';

.button {
	width: 85px;
  height: 45px;
	border: none;
	font-size: 1rem;
	cursor: pointer;
	transition: all 300ms ease-in-out;
}

.primary {
	background: var(--color__secondary);


}

.primary:hover {
	background: var(--color__secondary--light);
}

.primary:active {
	background: var(--color__secondary--dark);
}

.secondary {
	border: 3px solid var(--color__secondary);
	background: transparent;
	color: var(--color__secondary);

  /*composes: lol from '../../../stylesheets/colors.css';*/
}

.secondary:hover {
	border: 3px solid var(--color__secondary--light);
	color: var(--color__secondary--light);
}

.secondary:active {
	border: 3px solid var(--color__secondary--dark);
	background: transparent;
	color: var(--color__secondary--dark);
}
