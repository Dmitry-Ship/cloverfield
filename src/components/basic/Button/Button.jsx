import React, { PropTypes } from 'react';

import styles from './Button.scss';

import Icon from '../Icon';

const Button = ({
  type,
  label,
  onClick,
  className,
  kind,
  isLoading,
  iconName,
  disabled,
  children }) => (
    <button
      type={type}
      onClick={onClick}
      disabled={disabled}
      className={`${styles[kind]} ${className}`}
    >
      <div>
        {isLoading ? <Icon name="access_time" /> : <span>{label}</span>}
        {children}
        {iconName && <Icon name={iconName} />}
      </div>
    </button>
);

export default Button;

Button.defaultProps = {
  kind: 'primary',
  className: '',
  type: 'submit',
  disabled: false,
  children: null,
  iconName: null,
  label: 'Submit',
  onClick: null,
};

Button.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.oneOf(['submit', 'button', 'reset']),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  children: PropTypes.element,
  iconName: PropTypes.string,
  kind: PropTypes.oneOf(['primary', 'secondary']),
};
