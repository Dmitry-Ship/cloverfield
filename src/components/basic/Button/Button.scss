@value colors: "../../../styles/colors.scss";
@value accent, accent--light, accent--dark from colors;

.button {
  border: none;
  // box-sizing: padding-box;
  // display:block;
  font-size: 1rem;
  cursor: pointer;
  transition: 300ms ease-in-out;
  border-radius: 50px;
}

.loader,
.loader:after {
  border-radius: 50%;
  width: 10em;
  height: 10em;
}
.loader,
.loader:after {
  border-radius: 50%;
  width: 10em;
  height: 10em;
}

.loader {
  margin: auto;
  font-size: 3px;
  position: relative;
  border-top: 1.1em solid rgba(255, 255, 255, 0.2);
  border-right: 1.1em solid rgba(255, 255, 255, 0.2);
  border-bottom: 1.1em solid rgba(255, 255, 255, 0.2);
  border-left: 1.1em solid #ffffff;
  animation: spinner 1.1s infinite linear;
}
@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.primary {
  composes: button;
  background: accent;
  color: white;
  &:hover {
    background: accent--dark;
  }
}


.secondary {
  composes: button;
  background: transparent;
  border: 2px solid accent;
  color: accent;
  &:hover {
    background: accent;
    color: white;
  }
}

.normal {
  padding: 0 5%;
  height: 45px;
}

.small {
  padding: 0 10px;
  height: 32px;
}
