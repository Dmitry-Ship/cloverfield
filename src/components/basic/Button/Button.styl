@value colors "../../../styles/colors.styl";
@value accent from colors;
@value accent--light from colors;
@value accent--dark from colors;

.button
  border none
  width 85px
  height: 45px
  font-size 1rem
  cursor pointer
  transition 300ms ease-in-out

.primary
  composes button
  background accent
  color white
  &:hover
    background accent--dark

.secondary
  composes button
  background transparent
  border 3px solid accent
  color accent
  &:hover
    background: accent
    color white
