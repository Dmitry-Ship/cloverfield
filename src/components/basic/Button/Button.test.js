import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('<Button />', () => {
  it('should render initial state', () => {
    const tree = renderer.create(
      <Button />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display loader when isLoading is true', () => {
    const tree = renderer.create(
      <Button isLoading />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display an icon when icon name is provided', () => {
    const tree = renderer.create(
      <Button iconName="face" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display children when children are provided', () => {
    const tree = renderer.create(
      <Button >click me</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
