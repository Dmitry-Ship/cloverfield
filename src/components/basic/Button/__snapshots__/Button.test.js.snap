// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should display an icon when icon name is provided 1`] = `
<button
  className="primary normal "
  disabled={false}
  onClick={null}
  type="submit"
>
  Submit
  <i
    className=" material-icons"
  >
    face
  </i>
</button>
`;

exports[`<Button /> should display children when children are provided 1`] = `
<button
  className="primary normal "
  disabled={false}
  onClick={null}
  type="submit"
>
  click me
</button>
`;

exports[`<Button /> should display loader when isLoading is true 1`] = `
<button
  className="primary normal "
  disabled={false}
  onClick={null}
  type="submit"
>
  <div
    className="loader"
  />
</button>
`;

exports[`<Button /> should render initial state 1`] = `
<button
  className="primary normal "
  disabled={false}
  onClick={null}
  type="submit"
>
  Submit
</button>
`;
